addons/stock/report/report_package_barcode.xml

```diff
@@ -5,7 +5,7 @@
     <t t-call="web.basic_layout">
         <!-- quantity patterns are always 3 digit codes + 1 digit for number of digits (excluding units) -->
         <t t-set="uom_unit_id" t-value="env.ref('uom.product_uom_unit').id"/>
-        <t t-set="gs1_uom_patterns" t-value="{rule.associated_uom_id.id: rule.pattern[1:4] + str(len(str('{:.10f}'.format(rule.associated_uom_id.rounding).rstrip('0')).split('.')[1])) for rule in env['barcode.rule'].search([('associated_uom_id', '!=', False), ('associated_uom_id.id', '!=', uom_unit_id), ('is_gs1_nomenclature', '=', True)])}"/>
+        <t t-set="gs1_uom_patterns" t-value="{rule.associated_uom_id.id: rule.pattern[1:4] + str(env['decimal.precision'].precision_get('Product Unit of Measure') + 1) for rule in env['barcode.rule'].search([('associated_uom_id', '!=', False), ('associated_uom_id.id', '!=', uom_unit_id), ('is_gs1_nomenclature', '=', True)])}"/>
```

Comment:
Following the old formula, if the uom rounding was `0.01`, then the length of the decimal part would be `2`. Why would it be `3` now?